packages/api/lib/writeRecords/write-granules.js

```diff
@@ -896,6 +897,16 @@ const writeGranulesFromMessage = async ({
         knexOrTransaction: knex,
       });
 
+      // Check if granuleId exists across another collection
+      const granulesWithGranuleId = await getGranulesWithGranuleId(knex, postgresGranuleRecord.granule_id);
+      if (granulesWithGranuleId.some((g) => g.collection_cumulus_id !== postgresGranuleRecord.collection_cumulus_id)) {
+        log.error('Could not write granule. It already exists across another collection');
+        const conflictError = new Error(
+          `A granule already exists for granuleId: ${apiGranuleRecord.granuleId} with collectionId: ${apiGranuleRecord.collectionId}`
+        );
+        throw conflictError;
+      }
```

Comment:
I'm not sure where is the best place to put this logic. I added this here because `writeGranulesFromMessage` is the top most level we could make this check, but we could also put it in `_writeGranule`. The reason I didn't add this check to `_writeGranule` is because `writeGranuleFromApi` calls `_writeGranule` which would be redundant since the `granules POST` logic is updated to check for existing granules across all collections.

Reply:
We discussed this a lot in our architecture meetings and in huddles. https://wiki.earthdata.nasa.gov/pages/viewpage.action?pageId=271556967 Ultimately, this isn't the _ideal_ place for this logic but we are okay with it for this ticket since it meets the acceptance criteria, and we will revert this logic anyway. To make it easier to remove in the future, I added the `TODO: CUMULUS-3017` comment